### PR TITLE
Hyphy tools

### PR DESCRIPTION
**Pull request by stevenweaver**
This is a work in progress PR for getting the HyPhy tools IUC ready so that they can eventually be a part of usegalaxy.org.

* [x] -  update the tools to use HyPhy 2.5
* [x] -  confirm all tools meet the IUC standards
@stephenshank, @davebx and @stevenweaver, please use this branch (veg/tools-iuc/hyphy_tools) for any further development with the hyphy-tools that you want to eventually go into galaxy main.

This is related to galaxyproject/galaxy#7734

FOR CONTRIBUTOR:
* [x] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] - License permits unrestricted use (educational + commercial)
* [x] - This PR adds a new tool or tool collection
* [x] - This PR updates an existing tool or tool collection

### CODE REVIEW

**Comment by davebx** [comment]
@stevenweaver I think this is ready for review, do you agree? Although I'm not sure what bealign is doing in there.

**Comment by stevenweaver** [comment]
@davebx - absolutely. bealign only has formatting changes. When this is being reviewed, I'm more than willing to do whatever it takes to get this in. The longer it waits, the more work it will be for everyone involved.

File: tools/hyphy/hyphy_absrel.xml [hunk 1]
```diff
@@ -1,50 +1,43 @@
 <?xml version="1.0"?>
-<tool id="hyphy_absrel" name="HyPhy-aBSREL" version="@VERSION@+galaxy0">
-    <description>adaptive Branch Site Random Effects Likelihood</description>
+<tool id="hyphy_absrel" name="HyPhy-aBSREL" version="2.5.0+galaxy0">
+    <description>absrel</description>
```
Comment (bgruening) [inline tools/hyphy/hyphy_absrel.xml]: this description is less descriptive than the previous one ... and it is a repetition from the `name`

File: tools/hyphy/hyphy_absrel.xml [hunk 2]
```diff
@@ -1,50 +1,43 @@
...
-        > '$absrel_log'
-    #end if
-    ]]>
-    </command>
+    <expand macro="requirements"/>
+    <command detect_errors="exit_code"><![CDATA[
+        ln -s '$input_file' absrel_input.fa &&
+        ln -s '$input_nhx' absrel_input.nhx &&
+        @HYPHY_INVOCATION@ absrel
+            --alignment `pwd`/absrel_input.fa
```
Comment (bgruening) [inline tools/hyphy/hyphy_absrel.xml]: Do we need to the full path here? Can we not use `./`

File: tools/hyphy/hyphy_bgm.xml [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+    <description>- Detecting coevolving sites via Bayesian graphical models</description>
+    <macros>
+        <import>macros.xml</import>
+    </macros>
+    <expand macro="requirements"/>
+    <command detect_errors="exit_code"><![CDATA[
+        ln -s '$input_file' bgm_input.fa &&
+        ln -s '$input_nhx' bgm_input.nhx &&
+        @HYPHY_INVOCATION@ bgm
+            --alignment `pwd`/bgm_input.fa
```
Comment (bgruening) [inline tools/hyphy/hyphy_bgm.xml]: likewise ...

File: tools/hyphy/hyphy_gard.xml [hunk 4]
```diff
@@ -0,0 +1,119 @@
+<?xml version="1.0"?>
+<tool id="hyphy_gard" name="HyPhy-GARD" version="2.5.0+galaxy0">
+    <description>: Recombination Detection</description>
```
Comment (bgruening) [inline tools/hyphy/hyphy_gard.xml]: these `: ` is unusual. Just wanted to point this out, no strict need to remove it but usually, we do not do this.

File: tools/hyphy/hyphy_gard.xml [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+            #end if
+            --rv '$rate_cond.rate'
+            #if str($rate_cond.rate) != 'None':
+                --rate-classes '$rate_classes'
+            #end if
+            > '$gard_log'
+    ]]></command>
+    <inputs>
+        <param name="input_file" type="data" format="fasta" label="Input FASTA file"/>
+        <conditional name="datatype">
```
Comment (bgruening) [inline tools/hyphy/hyphy_gard.xml]: you have the same condition in hyphy-gbm, you have different value-names, but I think its the same condition ... maybe streamline this and move it to the macro file?

File: tools/hyphy/hyphy_meme.xml [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+            --code '$gencodeid'
+            --branches '$branches'
+            --pvalue '$p_value'
+            > '$meme_log'
+    ]]></command>
+    <inputs>
+        <expand macro="inputs"/>
+        <expand macro="gencode"/>
+        <expand macro="branches"/>
+        <param name="p_value" type="float" value=".1" min="0" max="1" label="P-value"/>
```
Comment (bgruening) [inline tools/hyphy/hyphy_meme.xml]: P-value threshold?

File: tools/hyphy/hyphy_relax.xml [hunk 7]
```diff
@@ -0,0 +1,53 @@
+<?xml version="1.0"?>
+<tool id="hyphy_relax" name="HyPhy-RELAX" version="2.5.0+galaxy0">
```
Comment (bgruening) [inline tools/hyphy/hyphy_relax.xml]: I think the version 2.5.0 should be a macro as well, it is everywhere.

File: tools/hyphy/hyphy_sm19.xml [hunk 8]
```diff
@@ -0,0 +1,82 @@
+<?xml version="1.0"?>
+<tool id="hyphy_sm19" name="HyPhy-SM2019" version="2.5.0+galaxy0">
+    <description>- Partition Tree using Modified Slatkin-Maddison Test</description>
```
Comment (bgruening) [inline tools/hyphy/hyphy_sm19.xml]: here you have a `- ` pattern and not a `: `

File: tools/hyphy/hyphy_absrel.xml (hunk 1 above)
Comment (davebx) [reply tools/hyphy/hyphy_absrel.xml]: Goodness, how did I miss that?

**Comment by davebx** [comment]
@bgruening thanks for looking, I believe I've fixed all the things now.

**Comment by bgruening** [comment]
Thanks a lot, @stevenweaver @davebx and all others involved! Great teamwork!

**Pull request merged**